engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/NodeDependency.scala

```diff
@@ -73,6 +74,14 @@ object DualParameterEditor {
  * to [[pl.touk.nussknacker.engine.definition.validator.ValidatorsExtractor]] which should decide whether new validator
  * should appear in configuration for certain parameter
  */
-@ConfiguredJsonCodec sealed trait ParameterValidator
+@ConfiguredJsonCodec sealed trait ParameterValidator {
 
-case object MandatoryValueValidator extends ParameterValidator
+  def isValid(expression: String) : Boolean
```

Comment:
WDYT about returning something like `Validated[ProcessCompilationError, Unit]`. We should minimize places that should be changed after adding new validator. One place that we will win is object `ParameterValidationError` and pattern matching inside factory method.